pkg/sql/pg_catalog.go

```diff
@@ -378,19 +378,37 @@ CREATE TABLE pg_catalog.pg_attribute (
 			// addColumn adds adds either a table or a index column to the pg_attribute table.
 			addColumn := func(column *sqlbase.ColumnDescriptor, attRelID tree.Datum, colID sqlbase.ColumnID) error {
 				colTyp := &column.Type
+				attTypMod := -1
+				width := int(colTyp.Width())
+				if width != 0 {
+					switch colTyp.Family() {
```

Comment:
should we add a `default: panic` or something here to make sure we don't miss other types that can have a width?

Reply:
I don't think so - these are exactly the set of types that need this logic, based on my investigations.